File: gut27/입국심사.py
#뭘 이분 탐색 해야하는지 잘 생각해보자
def solution(n, times):
    answer = 0
    l = 1
    r = max(times) * n
    
    while l <r:
        mid = (l + r) //2
        sum = 0
        for t in times:
            sum += mid // t
        if sum >= n:
            r = mid
        else:
            l = mid + 1
    answer = l
    return answer

File: Yunhyunjo/입국심사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

long long solution(int n, vector<int> times) {
    long long answer = 0;
    
    sort(times.begin(), times.end());
    long long lt = 1, rt = times[times.size() -1] * (long long)n, mid;
    
    while(lt <= rt){
        mid = (lt + rt) / 2;
        long long num = 0;
        for(int i: times){
            num = num + (mid / i);
        }
        if(num >= n) {
            rt = mid - 1;
            answer = mid;
        }
        else lt = mid + 1;
    }
    
    return answer;
}